pymess/models/common.py

```diff
@@ -18,3 +29,105 @@ def create_from_related_objects(self, *related_objects):
             self.create_from_related_object(related_object)
             for related_object in related_objects
         ]
+
+
+class BaseMessage(SmartModel):
+
+    STATE = ChoicesNumEnum()
+
+    sent_at = models.DateTimeField(verbose_name=_l('sent at'), null=True, blank=True, editable=False)
+    recipient = models.CharField(verbose_name=_l('recipient'), null=False, blank=False, max_length=20)
+    content = models.TextField(verbose_name=_l('content'), null=False, blank=False)
+    template_slug = models.SlugField(verbose_name=_l('slug'), max_length=100, null=True, blank=True, editable=False)
+    state = models.IntegerField(verbose_name=_l('state'), null=False, blank=False, choices=STATE.choices, editable=False)
+    backend = models.CharField(verbose_name=_l('backend'), null=True, blank=True, editable=False, max_length=250)
+    error = models.TextField(verbose_name=_l('error'), null=True, blank=True, editable=False)
+    extra_data = JSONField(verbose_name=_l('extra data'), null=True, blank=True, editable=False)
+    extra_sender_data = JSONField(verbose_name=_l('extra sender data'), null=True, blank=True, editable=False)
+    tag = models.SlugField(verbose_name=_l('tag'), null=True, blank=True, editable=False)
+
+    def __init__(self, *args, **kwargs):
+        assert len(self.STATE.choices) > 0, _l('STATE must be set properly')
+        super().__init__(*args, **kwargs)
+
+    @property
+    def failed(self):
+        return self.state == self.STATE.ERROR
+
+    def __str__(self):
+        return self.recipient
+
+    class Meta:
+        abstract = True
+        ordering = ('-created_at',)
+
+
+class PhoneNumberBasedRecipientMixin:
```

Comment:
bylo by vlastně lepší to dělat polem, které se samo normalizuje v clean metodě

Reply:
vratil som do povodneho stavu